src/shared/ml-thirdparty-client/index.ts

```diff
@@ -157,4 +158,123 @@ export class Client {
 
     // TODO: Implement communication with Mojaloop.
   }
+
+  /**
+   * Gets a list of PISP/DFSP participants
+   */
+  public async getParticipants(): Promise<void> {
+    if (this.simulator) {
+      // If the client is configured with a simulator, then it will not
+      // communicate with Mojaloop directly. Instead, it will only generate
+      // a random response that is injected to the internal routes.
+      return this.simulator.getParticipants()
+    }
+    // TODO: Add once implemented in sdk-standard components
+  }
+
+  /**
+   * Performs a request for a new consent
+   *
+   * @param requestBody         an consent request object as defined by the Mojaloop API.
+   * @param destParticipantId   ID of destination - to be used when sending request
+   */
+  public async postConsentRequests(
+    requestBody: SDKStandardComponents.PostConsentRequestsRequest,
+    destParticipantId: string
+  ): Promise<void> {
+    if (this.simulator) {
+      // If the client is configured with a simulator, then it will not
+      // communicate with Mojaloop directly. Instead, it will only generate
+      // a random response that is injected to the internal routes.
+      return this.simulator.postConsentRequests(requestBody)
+    }
+    await this.thirdpartyRequests.postConsentRequests(
+      requestBody,
+      destParticipantId
+    )
+  }
+
+  /**
+   * Performs a put request with authenticated consent request
+   *
+   * @param consentRequestId    unique identifier of the consent request
+   * @param requestBody         an object to authenticate consent as defined by the Mojaloop API.
+   * @param destParticipantId   ID of destination - to be used when sending request
+   */
+  public async putConsentRequests(
+    consentRequestId: string,
+    requestBody: SDKStandardComponents.PutConsentRequestsRequest,
+    destParticipantId: string
+  ): Promise<void> {
+    if (this.simulator) {
+      // If the client is configured with a simulator, then it will not
+      // communicate with Mojaloop directly. Instead, it will only generate
+      // a random response that is injected to the internal routes.
+      return this.simulator.putConsentRequests(consentRequestId, requestBody)
+    }
+    await this.thirdpartyRequests.putConsentRequests(
```

Comment:
Why await here? Isn't that just blocking and then returning nothing? We shouldn't await unless we need the result for something else. If it's the end of a function, we should return the promise and allow the caller to decide whether to await or not...

Reply:
I think we still need to perform the await here to get back the result of the request (i.e., the response code). 
However, in the current implementation, as we still haven't handled the error cases, it's just awaiting for the response without doing anything. I personally feel that the function still need to run asynchronously and return a future instead of just invoking the async function and return directly.